Source/Core/VideoBackends/D3D/D3DState.h

```diff
@@ -224,43 +225,27 @@ class StateManager
 private:
   enum DirtyFlags
   {
-    DirtyFlag_Texture0 = 1 << 0,
-    DirtyFlag_Texture1 = 1 << 1,
-    DirtyFlag_Texture2 = 1 << 2,
-    DirtyFlag_Texture3 = 1 << 3,
-    DirtyFlag_Texture4 = 1 << 4,
-    DirtyFlag_Texture5 = 1 << 5,
-    DirtyFlag_Texture6 = 1 << 6,
-    DirtyFlag_Texture7 = 1 << 7,
-
-    DirtyFlag_Sampler0 = 1 << 8,
-    DirtyFlag_Sampler1 = 1 << 9,
-    DirtyFlag_Sampler2 = 1 << 10,
-    DirtyFlag_Sampler3 = 1 << 11,
-    DirtyFlag_Sampler4 = 1 << 12,
-    DirtyFlag_Sampler5 = 1 << 13,
-    DirtyFlag_Sampler6 = 1 << 14,
-    DirtyFlag_Sampler7 = 1 << 15,
-
-    DirtyFlag_PixelConstants = 1 << 16,
-    DirtyFlag_VertexConstants = 1 << 17,
-    DirtyFlag_GeometryConstants = 1 << 18,
-
-    DirtyFlag_VertexBuffer = 1 << 19,
-    DirtyFlag_IndexBuffer = 1 << 20,
-
-    DirtyFlag_PixelShader = 1 << 21,
-    DirtyFlag_VertexShader = 1 << 22,
-    DirtyFlag_GeometryShader = 1 << 23,
-
-    DirtyFlag_InputAssembler = 1 << 24,
-    DirtyFlag_BlendState = 1 << 25,
-    DirtyFlag_DepthState = 1 << 26,
-    DirtyFlag_RasterizerState = 1 << 27,
-    DirtyFlag_Framebuffer = 1 << 28
+    DirtyFlag_Texture0 = 0,
+    DirtyFlag_Sampler0 = VideoCommon::MAX_PIXEL_SHADER_SAMPLERS,
+    DirtyFlag_PixelConstants = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 1,
+    DirtyFlag_VertexConstants = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 2,
+    DirtyFlag_GeometryConstants = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 3,
+
+    DirtyFlag_VertexBuffer = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 4,
+    DirtyFlag_IndexBuffer = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 5,
+
+    DirtyFlag_PixelShader = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 6,
+    DirtyFlag_VertexShader = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 7,
+    DirtyFlag_GeometryShader = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 8,
+
+    DirtyFlag_InputAssembler = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 9,
+    DirtyFlag_BlendState = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 10,
+    DirtyFlag_DepthState = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 11,
+    DirtyFlag_RasterizerState = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 12,
+    DirtyFlag_Framebuffer = (VideoCommon::MAX_PIXEL_SHADER_SAMPLERS * 2) + 13
```

Comment:
I think this would look better as
```cpp
DirtyFlag_Texture0 = 0,
DirtyFlag_Sampler0 = DirtyFlag_Texture0 + VideoCommon::MAX_PIXEL_SHADER_SAMPLERS,
DirtyFlag_PixelConstants = DirtyFlag_Sampler0 + VideoCommon::MAX_PIXEL_SHADER_SAMPLERS,
DirtyFlag_VertexConstants,
DirtyFlag_GeometryConstants,
...
```

Unless you need to keep the bit layout from changing for some reason (e.g. abi, serialization), it's much nicer to let the compiler assign values.